packages/web3-eth/types/tests/eth.tests.ts

```diff
@@ -0,0 +1,414 @@
+});
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, 100);
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, '100');
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, (error: Error, data: string) => {});
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, '100', (error: Error, data: string) => {});
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, 100, (error: Error, data: string) => {});
+
+// $ExpectType Promise<string>
+eth.call({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe", // contract address
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, 100, (error: Error, data: string) => {});
+
+// $ExpectType Promise<number>
+eth.estimateGas({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe",
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+});
+// $ExpectType Promise<number>
+eth.estimateGas({
+    to: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe",
+    data: "0xc6888fa10000000000000000000000000000000000000000000000000000000000000003"
+}, (error: Error, gas: number) => {});
+
+// $ExpectType Promise<Log[]>
+eth.getPastLogs({
+    address: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe",
+    topics: ["0x033456732123ffff2342342dd12342434324234234fd234fd23fd4f23d4234"]
+});
+// $ExpectType Promise<Log[]>
+eth.getPastLogs({
+    address: "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe",
+    topics: ["0x033456732123ffff2342342dd12342434324234234fd234fd23fd4f23d4234"]
+}, (error: Error, logs: Log[]) => {});
+
+// $ExpectType Promise<string[]>
+eth.getWork();
+// $ExpectType Promise<string[]>
+eth.getWork((error: Error, result: string[]) => {});
+
+// $ExpectType Promise<boolean>
+eth.submitWork([
+    "0x0000000000000001",
+    "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
+    "0xD1FE5700000000000000000000000000D1FE5700000000000000000000000000"
+]);
+
+// $ExpectType Promise<boolean>
+eth.submitWork([
+    "0x0000000000000001",
+    "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
+    "0xD1FE5700000000000000000000000000D1FE5700000000000000000000000000"
+], (error: Error, result: boolean) => {});
```

Comment:
Idk, but maybe it would be better to split that up in some files. It's easier to fix something and to keep the overview if this would be splitted up. :)

Reply:
it is all contained within the same class so in my eyes it should be in 1. You have to generate the class and test the context of it, splitting up feels unnecessary. I could split it up and import them into the the 1 test file for example 

in eth.tests.ts
```
import { SignedTests }  from './signed-transaction-test.ts'

const eth = new Eth('http://localhost:8545');

SignedTests(eth)
..... rest to follow
```

would you like that more?

Reply:
Yeah that's true let us keep it in one file :)